Tests/TuistGeneratorTests/Linter/StaticProductsGraphLinterTests.swift

```diff
@@ -1289,6 +1289,60 @@ class StaticProductsGraphLinterTests: XCTestCase {
         XCTAssertEqual(results, [])
     }
 
+    func test_lint_whenMacrosLinkedTwice() throws {
```

Comment:
thanks for adding this test 👍 

If I followed along this use case the static framework is linked twice!

![example](https://github.com/tuist/tuist/assets/11914919/c0059377-3c1d-433f-9cd6-32f4c3cf1aab)

Mind elaborating on how this can be a false positive?

Reply:
In our case, our `MacroSupport.xcframework` is dynamic:
`../MacroSupport.framework/MacroSupport: Mach-O 64-bit dynamically linked shared library arm64` 

This is our graph dependency for MacroSupport

![image](https://github.com/user-attachments/assets/f2f61fce-f2dc-44fd-a212-97e5233a5095)

Normally for static frameworks linked to multiple targets I would see duplicate symbols at run time, for example: `objc[13219]: Class _TtC13TestFramework8Foo is implemented in both...` but in this case I don't see any symbol collisions.

Despite this in tuist we see a long list of warnings like the ones mentioned in the [bug](https://github.com/tuist/tuist/issues/6209). This case may be different, since our macro framework is dynamic and not static, but it seems to be closely related and I haven't found anything wrong yet.

However, I am not familiar with how custom macros and SwiftXX.xcframeworks are processed at compile time. Can there be collisions between macros? And if so, would Xcode be able to warn us like it does with symbols?

Reply:
@kwridan the false positive is from the fact that the macro -> swift syntax dependency shouldn't be included in linking logic.  the dependency on the macro is not a linking dependency so what the macro links against has no bearing on the rest of the dependency graph.

Reply:
> the false positive is from the fact that the macro -> swift syntax dependency shouldn't be included in linking logic

That makes sense. However, shouldn't [this](https://github.com/tuist/tuist/pull/6210/files#diff-ff24d355cfe64c86061e6d2562b749e5ac160862dc21788fb7bc3d46d3235e0aR143-R144) only skip links where a target depends on a macro directly? Instead of skipping any links that depend on a `.macro` target transitively?

Reply:
Thanks for the clarification @waltflanagan - Indeed anything below the macro executable target shouldn't be taken intro consideration 👍 

> the false positive is from the fact that the macro -> swift syntax dependency shouldn't be included in linking logic.

I believe that's the case already without the changes in the PR. I think the missing piece which I wasn't aware of was in this example, `MacroStaticFramework` is part of the macro package.

![Screenshot 2024-08-08 at 09 10 39](https://github.com/user-attachments/assets/6439fc30-10f3-4147-95ba-e4d00f51d7b8)

I had initially thought it was only Macro executable and below. Just re-read up on Macros and realised the declarations for the Macro signatures live in libraries.

> That makes sense. However, shouldn't this only skip links where a target depends on a macro directly? Instead of skipping any links that depend on a .macro target transitively?

Indeed the changes in this PR would now fail to identify real issues such as 

![Screenshot 2024-08-08 at 10 12 49](https://github.com/user-attachments/assets/5dde9cd4-c5f7-42b5-80d6-1d2715c6872a)

Provided the findings above are correct, I believe what we need is something along the lines of to skip macro signature libraries:

```swift   
        if case let .target(name, path) = staticProduct,
            !graphTraverser.directSwiftMacroExecutables(path: path, name: name).isEmpty {
            return []
        }
```